BlueCapKit/Central/Peripheral.swift

```diff
@@ -232,6 +232,30 @@ public class Peripheral: NSObject, CBPeripheralDelegate {
         self.cbPeripheral.delegate = nil
     }
 
+    func updateDiscovery(_ peripheral: CBPeripheralInjectable, advertisements: [String : Any], RSSI: Int) throws {
+        guard self.identifier == peripheral.identifier else {
+            throw PeripheralError.invalidCbPeripheral
+        }
+
+        if let oldCbPeripheral = self.cbPeripheral as? CBPeripheral,
+            let aCbPeripehral = peripheral as? CBPeripheral {
+            if oldCbPeripheral !== aCbPeripehral {
```

Comment:
This does not need to be a separate if statement. It can be combined with the outer if.